python/cudf/cudf/tests/test_copying.py

```diff
@@ -52,3 +53,395 @@ def test_null_copy():
     col = Series(np.arange(2049))
     col[:] = None
     assert len(col) == 2049
+
+
+# TODO: Make use of set_option context manager
+# once https://github.com/rapidsai/cudf/issues/12736
+# is resolved.
+
+
+def test_series_setitem_cow_on():
+    original_cow_setting = cudf.get_option("copy_on_write")
+    cudf.set_option("copy_on_write", True)
+    actual = cudf.Series([1, 2, 3, 4, 5])
+    new_copy = actual.copy(deep=False)
+
+    actual[1] = 100
+    assert_eq(actual, cudf.Series([1, 100, 3, 4, 5]))
+    assert_eq(new_copy, cudf.Series([1, 2, 3, 4, 5]))
+
+    cudf.set_option("copy_on_write", original_cow_setting)
+
+
+def test_series_setitem_cow_off():
+    original_cow_setting = cudf.get_option("copy_on_write")
+    cudf.set_option("copy_on_write", False)
+    actual = cudf.Series([1, 2, 3, 4, 5])
+    new_copy = actual.copy(deep=False)
+
+    actual[1] = 100
+    assert_eq(actual, cudf.Series([1, 100, 3, 4, 5]))
+    assert_eq(new_copy, cudf.Series([1, 100, 3, 4, 5]))
+    cudf.set_option("copy_on_write", original_cow_setting)
+
```

Comment:
It might be worth writing a small context manager for use with these tests now, and then replacing them once #12736 is addressed for the more generic `option` solution. Up to you, though.
```suggestion
@contextlib.contextmanager
def with_copy_on_write(on):
    original_cow_setting = cudf.get_option("copy_on_write")
    cudf.set_option("copy_on_write", on)
    yield
    cudf.set_option("copy_on_write", original_cow_setting)

@with_copy_on_write(on=False)
def test_series_setitem_cow_off():
    actual = cudf.Series([1, 2, 3, 4, 5])
    new_copy = actual.copy(deep=False)

    actual[1] = 100
    assert_eq(actual, cudf.Series([1, 100, 3, 4, 5]))    
```

Reply:
Done 👍  Less lines of code too 😉